sktime/clustering/spatio_temporal/_st_dbscan.py

```diff
@@ -0,0 +1,296 @@
+"""
+ST-DBSCAN - fast scalable implementation of ST DBSCAN
+            scales also to memory by splitting into frames
+            and merging the clusters together
+"""
+
+__author__ = ["VectorNd"]
+
+from sktime.clustering import BaseClusterer
+
+import numpy as np
+
+from scipy.spatial.distance import pdist, squareform
+from scipy.sparse import coo_matrix
+
+from sklearn.cluster import DBSCAN
+from sklearn.utils import check_array
+from sklearn.neighbors import NearestNeighbors
+
+import warnings
+
+class ST_DBSCAN(BaseClusterer):
+    """
+    A class to perform the ST_DBSCAN clustering
+    Parameters
+    ----------
+    eps1 : float, default=0.5
+        The spatial density threshold (maximum spatial distance) between 
+        two points to be considered related.
+    eps2 : float, default=10
+        The temporal threshold (maximum temporal distance) between two 
+        points to be considered related.
+    min_samples : int, default=5
+        The number of samples required for a core point.
+    metric : string default='euclidean'
+        The used distance metric - more options are
+        'braycurtis', 'canberra', 'chebyshev', 'cityblock', 'correlation',
+        'cosine', 'dice', 'euclidean', 'hamming', 'jaccard', 'jensenshannon',
+        'kulsinski','manhattan'.
+    n_jobs : int or None, default=-1
+        The number of processes to start -1 means use all processors 
+    Attributes
+    ----------
+    labels : array, shape = [n_samples]
+        Cluster labels for the data - noise is defined as -1
+    """
+
+    _tags = {
+        "authors": ["VectorNd","eren-ck"],  
+        "python_dependencies": ["scipy","sklearn"],  
+        "X_inner_mtype": "numpyflat",
+        "capability:multivariate": False,
+        "capability:unequal_length": False,
+        "capability:missing_values": False,
+        "capability:multithreading": False,
+        "capability:predict": True,  
+        "capability:predict_proba": False, 
+        "capability:out_of_sample": True, 
+    }
+
+    def __init__(self, eps1=0.5, eps2=10, min_samples=5, metric='euclidean',n_jobs=-1):
+        self.eps = eps1
+        self.eps1 = eps1
+        self.eps2 = eps2
+        self.min_samples = min_samples
+        self.metric = metric
+        self.n_jobs = n_jobs
+
+        super().__init__()
+
+    def _fit(self, X):
+        """Apply the ST DBSCAN algorithm 
+        ----------
+        X : 2D numpy array with
+            The first element of the array should be the time 
+            attribute as float. The following positions in the array are 
+            treated as spatial coordinates. The structure should look like this [[time_step1, x, y], [time_step2, x, y]..]
+            For example 2D dataset:
+            array([[0,0.45,0.43],
+            [0,0.54,0.34],...])
+        Returns
+        -------
+        self
+        """
+
+        X = check_array(X)
```

Comment:
check_array is not needed in sktime, the boilerplate will do checking and conversion already.